pandas/core/indexes/base.py

```diff
@@ -1105,6 +1105,52 @@ def _format_attrs(self):
         """
         return format_object_attrs(self)
 
+    def to_index(self, sep=None):
+        """
+        Convert a MultiIndex to an Index of Tuples containing the level values.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        sep : str, optional
+            Not yet implemented.
+
+        Returns
+        -------
+        pd.Index : an Index with the MultiIndex data represented in Tuples.
+
+        Notes
+        -----
+        This method will simply return the caller if called by anything other
+        than a MultiIndex.
+
+        See Also
+        --------
+        to_series : Similar method to construct a Series.
+        to_frame : Similar method to construct a DataFrame.
+
+        Examples
+        --------
+        >>> index = pd.MultiIndex.from_product(
+        ...     [['foo', 'bar'], ['baz', 'qux']],
+        ...     names=['a', 'b'])
+        >>> index.to_index()
+        Index([('foo', 'baz'), ('foo', 'qux'),
+               ('bar', 'baz'), ('bar', 'qux')],
+              dtype='object')
+        """
+        if sep is not None:
+            # TODO: Add support for separator to return strs instad of tuples
+            raise NotImplementedError
+
+        if not isinstance(self, ABCMultiIndex):
+            return self
+
+        idx = Index(self.values, tupleize_cols=False)
```

Comment:
Maybe ``iter(self)`` is slightly more efficient? (as I think it happens anyway inside ``self.values``, and I don't think passing through an ``array`` helps)

Reply:
Sorry, I had forgot this... what do you @WillAyd think?

Reply:
If use don't do it here, the `Index` constructors has to convert tot an array anyway for storage, right?

```python
In [4]: pd.Index(np.array([(1, 2), (3, 4)], dtype=object), tupleize_cols=False).values
Out[4]: array([(1, 2), (3, 4)], dtype=object)
```

In theory, the array of tuples should just pass through without copy.

Reply:
@TomAugspurger You're right

``` python
In [2]: a = np.array([1, 2], dtype=object)

In [3]: a[:] = [(1, 2), (3, 4)]

In [4]: id(a), id(pd.Index(a, tupleize_cols=False).values.base)
Out[4]: (140686849540864, 140686849540864)
```
sorry for the noise